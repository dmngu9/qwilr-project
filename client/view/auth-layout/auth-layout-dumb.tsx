import * as React from 'react';

import { Background, FormSection } from './styles';
import { withRouteLayout } from '../with-route-layout';

interface Props {
    children?: React.ReactNode;
}

export const AuthLayoutDumb: React.StatelessComponent<Props> = ({ children }) => (
    <Background>
        <FormSection>{children}</FormSection>
    </Background>
);

export default withRouteLayout(AuthLayoutDumb);
