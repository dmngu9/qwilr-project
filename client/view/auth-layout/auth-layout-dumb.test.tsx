import * as React from 'react';
import { shallow } from 'enzyme';

import { AuthLayoutDumb } from './auth-layout-dumb';

describe('<AuthLayoutDumb/>', () => {
    it('should render correctly without children', () => {
        const wrapper = shallow(<AuthLayoutDumb />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render correctly with children', () => {
        const wrapper = shallow(
            <AuthLayoutDumb>
                <div>Children</div>
            </AuthLayoutDumb>
        );

        expect(wrapper).toMatchSnapshot();
    });
});
