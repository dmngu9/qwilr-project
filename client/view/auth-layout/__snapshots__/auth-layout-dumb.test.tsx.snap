// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthLayoutDumb/> should render correctly with children 1`] = `
<styled.div>
  <styled.div>
    <div>
      Children
    </div>
  </styled.div>
</styled.div>
`;

exports[`<AuthLayoutDumb/> should render correctly without children 1`] = `
<styled.div>
  <styled.div />
</styled.div>
`;
